From df545b9db6197922c4a5d96e44b126676d2672dc Mon Sep 17 00:00:00 2001
From: Daniele Palmas <dnlplm@gmail.com>
Date: Wed, 5 Jul 2023 15:13:29 +0200
Subject: [PATCH 13/17] drivers: bus: mhi: host: add error print in
 prepare_for_power_up

Add missing print in a few error paths, useful for debugging.

Signed-off-by: Daniele Palmas <dnlplm@gmail.com>
---
 drivers/bus/mhi/host/init.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/bus/mhi/host/init.c b/drivers/bus/mhi/host/init.c
index 9ca7339dfe56..0d81328f3b75 100644
--- a/drivers/bus/mhi/host/init.c
+++ b/drivers/bus/mhi/host/init.c
@@ -1127,8 +1127,10 @@ int mhi_prepare_for_power_up(struct mhi_controller *mhi_cntrl)
 	mutex_lock(&mhi_cntrl->pm_mutex);
 
 	ret = mhi_init_dev_ctxt(mhi_cntrl);
-	if (ret)
+	if (ret) {
+		dev_err(dev, "Unable to init dev context\n");
 		goto error_dev_ctxt;
+	}
 
 	ret = mhi_read_reg(mhi_cntrl, mhi_cntrl->regs, BHIOFF, &bhi_off);
 	if (ret) {
@@ -1179,6 +1181,7 @@ int mhi_prepare_for_power_up(struct mhi_controller *mhi_cntrl)
 			ret = mhi_rddm_prepare(mhi_cntrl,
 					       mhi_cntrl->rddm_image);
 			if (ret) {
+				dev_err(dev, "Error mhi_rddm_prepare\n");
 				mhi_free_bhie_table(mhi_cntrl,
 						    mhi_cntrl->rddm_image);
 				goto error_reg_offset;
-- 
2.43.2

